agent/hippo4j-agent-core/src/main/java/cn/hippo4j/agent/core/registry/AgentThreadPoolRegistry.java

```diff
@@ -26,40 +26,36 @@
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ThreadPoolExecutor;
 
-public class AgentThreadPoolInstanceRegistry {
+public class AgentThreadPoolRegistry {
 
-    private static final ILog LOGGER = LogManager.getLogger(AgentThreadPoolInstanceRegistry.class);
+    private static final ILog LOGGER = LogManager.getLogger(AgentThreadPoolRegistry.class);
 
-    private final Map<String, AgentThreadPoolExecutorHolder> holderMap = new ConcurrentHashMap<>();
+    private static final Map<String, AgentThreadPoolExecutorHolder> HOLDER_MAP = new ConcurrentHashMap<>();
 
-    public final Map<ThreadPoolExecutor, Class<?>> earlyConstructMap = new ConcurrentHashMap<>();
+    private static final Map<ThreadPoolExecutor, Class<?>> EARLY_CONSTRUCT_MAP = new ConcurrentHashMap<>();
 
-    private volatile static AgentThreadPoolInstanceRegistry INSTANCE;
+    public static Map<String, AgentThreadPoolExecutorHolder> getHolderMap() {
+        return HOLDER_MAP;
+    }
 
-    private AgentThreadPoolInstanceRegistry() {
+    public static void putHolder(String executorName, ThreadPoolExecutor executor, Properties properties) {
+        AgentThreadPoolExecutorHolder holder = new AgentThreadPoolExecutorHolder(executorName, executor, properties);
+        HOLDER_MAP.put(executorName, holder);
     }
 
-    public static AgentThreadPoolInstanceRegistry getInstance() {
-        if (INSTANCE == null) {
-            synchronized (AgentThreadPoolInstanceRegistry.class) {
-                if (INSTANCE == null) {
-                    INSTANCE = new AgentThreadPoolInstanceRegistry();
-                }
-            }
-        }
-        return INSTANCE;
+    public static AgentThreadPoolExecutorHolder getHolder(String executorName) {
+        return Optional.ofNullable(HOLDER_MAP.get(executorName)).orElse(AgentThreadPoolExecutorHolder.EMPTY);
     }
 
-    public Map<String, AgentThreadPoolExecutorHolder> getHolderMap() {
-        return holderMap;
+    public static Map<ThreadPoolExecutor, Class<?>> getEarlyConstructMap() {
+        return EARLY_CONSTRUCT_MAP;
     }
 
-    public void putHolder(String executorName, ThreadPoolExecutor executor, Properties properties) {
-        AgentThreadPoolExecutorHolder holder = new AgentThreadPoolExecutorHolder(executorName, executor, properties);
-        holderMap.put(executorName, holder);
+    public static void putEarlyConstruct(ThreadPoolExecutor executor, Class<?> earlyConstruct) {
+        EARLY_CONSTRUCT_MAP.put(executor, earlyConstruct);
```

Comment:
I think two optimizations need to be done
- No need for function wrapping via `putEarlyConstruct`
- Please rename `earlyConstruct` in the parameter list to `referencedClass`

Reply:
Good idea. I'll try to finish it later